src/uu/ls/src/ls.rs

```diff
@@ -1128,18 +1204,41 @@ fn get_system_time(md: &Metadata, config: &Config) -> Option<SystemTime> {
     }
 }
 
-fn get_time(md: &Metadata, config: &Config) -> Option<time::Tm> {
-    let duration = get_system_time(md, config)?
-        .duration_since(UNIX_EPOCH)
-        .ok()?;
-    let secs = duration.as_secs() as i64;
-    let nsec = duration.subsec_nanos() as i32;
-    Some(time::at(Timespec::new(secs, nsec)))
+fn get_time(md: &Metadata, config: &Config) -> Option<chrono::DateTime<chrono::Local>> {
+    let time = get_system_time(md, config)?;
+    Some(time.into())
 }
 
 fn display_date(metadata: &Metadata, config: &Config) -> String {
     match get_time(metadata, config) {
-        Some(time) => strftime("%F %R", &time).unwrap(),
+        Some(time) => {
+            //Date is recent if from past 6 months
+            //According to GNU a Gregorian year has 365.2425 * 24 * 60 * 60 == 31556952 seconds on the average.
+            //https://github.com/coreutils/coreutils/blob/master/src/ls.c#L4385
+            let recent = time + chrono::Duration::seconds(31556952 / 2) > chrono::Local::now();
+
+            //For reference see https://github.com/coreutils/coreutils/blob/master/src/ls.c#L2416
+            match config.time_style {
+                TimeStyle::FullIso => time.format("%Y-%m-%d %H:%M:%S.%N %z"),
+                TimeStyle::LongIso => time.format("%Y-%m-%d %H:%M"),
+                TimeStyle::Iso => time.format(if recent {
+                    "%Y-%m-%d %H:%M"
```

Comment:
> This is not the format I'm getting on my machine. I get the recent files formatted like this:
> 
> ```
> 11-25 14:54 (%m-%d %H:%M)
> ```

Yep, that's a mistake. 



> Also, we currently don't do anything clever in the long format to ensure everything is lined up, so I think this might mess up the spacing a little bit?

I'll look into padding that.

Reply:
I don't think the padding is an issue but you could QA that for me if I finish this up